src/torchmetrics/functional/audio/srmr.py

```diff
@@ -0,0 +1,297 @@
+# Copyright The Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Note: without special mention, the functions in this file are mainly translated from
+# the SRMRpy package for batched processing with pytorch
+
+from functools import lru_cache
+from math import ceil
+from typing import Optional, Tuple
+
+import torch
+from torch import Tensor
+from torch.nn.functional import pad
+
+from torchmetrics.utilities import rank_zero_warn
+from torchmetrics.utilities.imports import _GAMMATONE_AVAILABEL, _TORCHAUDIO_AVAILABEL
+
+if _TORCHAUDIO_AVAILABEL:
+    from torchaudio.functional.filtering import lfilter
+else:
+    lfilter = None
+    __doctest_skip__ = ["speech_reverberation_modulation_energy_ratio"]
+
+if _GAMMATONE_AVAILABEL:
+    from gammatone.fftweight import fft_gtgram
+    from gammatone.filters import centre_freqs, make_erb_filters
```

Comment:
are these missing in `torchaudio`?

Reply:
Yes, torchaudio doesn't implements these functions.

Reply:
Or what do you think about moving the relevant code into torchmetrics

Reply:
if there are dependency issues due to `gammatone` not being maintained for 9 years, then yes we should include them in tm and just refer to `gammatone`. If not, then lets just keep this dependency.

Reply:
I am opening a [pull request](https://github.com/jfsantos/SRMRpy/pull/14) in the upstream trying to fix the [dependency issue](https://github.com/Lightning-AI/torchmetrics/actions/runs/5279262983/jobs/9549684134?pr=1792).

Reply:
The [pull request](https://github.com/jfsantos/SRMRpy/pull/14) has not be merged by the author, so the `gammatone` in requirements is removed temporarily and will be added when it is merged.